gym/lib/gym/options.rb

```diff
@@ -103,6 +103,16 @@ def self.plain_options
                                          UI.important "Using legacy build system - waiting for radar to be fixed: https://openradar.appspot.com/radar?id=4952000420642816"
                                        end
                                      end),
+        FastlaneCore::ConfigItem.new(key: :use_generic_archive_fix,
+                                    env_name: "GYM_USE_GENERIC_ARCHIVE_FIX",
+                                    description: "Use fix for generic arhives https://github.com/CocoaPods/CocoaPods/issues/4178",
+                                    default_value: false,
+                                    is_string: false,
+                                    verify_block: proc do |value|
+                                      if value
+                                        UI.important "Using generic archive fix - waiting for issue to be fixed: https://github.com/CocoaPods/CocoaPods/issues/4178"
+                                      end
+                                    end),
```

Comment:
Because the hope is that this is a temporary work-around, we should avoid making a `ConfigItem` for this... since it would become a permanent part of the API.

Let's just use an ENV variable feature switch for now, like we started sketching out here https://github.com/fastlane/fastlane/pull/4066/files#diff-6a87cb42f5a457351cff9096fe9b12ceR77